# Revolute_55_position_controller:
#   type: effort_controllers/JointPositionController
#   joint: Revolute_55
#   pid: {p: 1.0, i: 0.01, d: 0.10}

# Revolute_56_position_controller:
#   type: effort_controllers/JointPositionController
#   joint: Revolute_56
#   pid: {p: 1.0, i: 0.01, d: 0.10}
  
# Revolute_57_position_controller:
#   type: effort_controllers/JointPositionController
#   joint: Revolute_57
#   pid: {p: 1.0, i: 0.01, d: 0.10}
  
# Revolute_58_position_controller:
#   type: effort_controllers/JointPositionController
#   joint: Revolute_58
#   pid: {p: 1.0, i: 0.01, d: 0.10}

# Revolute_55_position_controller:
#   type: velocity_controllers/JointVelocityController
#   joint: Revolute_55
#   pid: {p: 100.0, i: 0.01, d: 0.10}
  

# Revolute_56_position_controller:
#   type: velocity_controllers/JointVelocityController
#   joint: Revolute_56
#   pid: {p: 100.0, i: 0.01, d: 0.10}
  
  
# Revolute_57_position_controller:
#   type: velocity_controllers/JointVelocityController
#   joint: Revolute_57
#   pid: {p: 100.0, i: 0.01, d: 0.10}
  
  
# Revolute_58_position_controller:
#   type: velocity_controllers/JointVelocityController
#   joint: Revolute_58
#   pid: {p: 100.0, i: 0.01, d: 0.10}


  
  
Revolute_59_position_controller:
  type: effort_controllers/JointPositionController
  joint: Revolute_59
  pid: {p: 100.0, i: 0.01, d: 0.10}
  
Revolute_60_position_controller:
  type: effort_controllers/JointPositionController
  joint: Revolute_60
  pid: {p: 100.0, i: 0.01, d: 0.10}

/gazebo_ros_control:
  pid_gains:
    Revolute_55:
      p: 0.15
      i: 0.0
      d: 0.0

    Revolute_56:
      p: 0.15
      i: 0.0
      d: 0
    Revolute_57:
      p: 0.15
      i: 0.0
      d: 0.0

    Revolute_58:
      p: 0.15
      i: 0.0
      d: 0.0



mobile_base_controller:
  type: "diff_drive_controller/DiffDriveController"
  left_wheel : 'Revolute_55'
  right_wheel: 'Revolute_56'
  publish rate: 50.0
  pose_covariance_diagonal : [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
  twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
  wheel_separation : 1.0
  wheel_radius : 0.3
  wheel_separation_multiplier: 1.0
  wheel_radius_multiplier    : 1.0 
  cmd_vel_timeout: 0.25
  base_frame_id: base_footprint
  linear:
    x:
      has_velocity_limits    : true
      max_velocity           : 1.0  
      min_velocity           : 0.0
      has_acceleration_limits: true
      max_acceleration       : 0.8  
      min_acceleration       : 0.0
      has_jerk_limits        : true
      max_jerk               : 5.0  
  angular:
    z:
      has_velocity_limits    : true
      max_velocity           : 1.7  
      has_acceleration_limits: true
      max_acceleration       : 1.5  
      has_jerk_limits        : true
      max_jerk               : 2.5  


mobile_base_controller1:
  type: "diff_drive_controller/DiffDriveController"
  left_wheel : 'Revolute_57'
  right_wheel: 'Revolute_58'
  publish rate: 50.0
  pose_covariance_diagonal : [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
  twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
  wheel_separation : 1.0
  wheel_radius : 0.3
  wheel_separation_multiplier: 1.0
  wheel_radius_multiplier    : 1.0 
  cmd_vel_timeout: 0.25
  base_frame_id: base_footprint
  linear:
    x:
      has_velocity_limits    : true
      max_velocity           : 1.0  
      min_velocity           : 0.0
      has_acceleration_limits: true
      max_acceleration       : 0.8  
      min_acceleration       : 0.0
      has_jerk_limits        : true
      max_jerk               : 5.0  
  angular:
    z:
      has_velocity_limits    : true
      max_velocity           : 1.7  
      has_acceleration_limits: true
      max_acceleration       : 1.5  
      has_jerk_limits        : true
      max_jerk               : 2.5  

  


